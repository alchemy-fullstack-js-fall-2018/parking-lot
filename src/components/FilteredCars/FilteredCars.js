import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import Cars from '../../components/Cars/Cars';

export default class FilteredCars extends PureComponent {
  static propTypes = {
    cars: PropTypes.array.isRequired,
    searchTerm: PropTypes.string.isRequired,
    updateSearchTerm: PropTypes.func.isRequired,
    summary: PropTypes.string
  };

  render() {
    const { cars, searchTerm, updateSearchTerm, summary } = this.props;
    return (
      <Fragment>
        <form>
          <label htmlFor="searchTerm">Search: </label>
          <input type="text" value={searchTerm} onChange={updateSearchTerm} />
        </form>
        <Cars cars={cars} summary={summary}/>
      </Fragment>
    );
  }
}
