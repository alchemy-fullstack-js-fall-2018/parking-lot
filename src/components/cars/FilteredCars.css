.header {
  text-align: center;
}

.form {
  display: block;
  text-align: center;
}