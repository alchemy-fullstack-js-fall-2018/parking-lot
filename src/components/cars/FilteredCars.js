import React, { Fragment } from 'react';
import styles from './FilteredCars.css';
import Cars from '../../components/cars/Cars';

export default function FilteredCars({ cars, searchTerm, updateSearchTerm }) {
  return (
    <Fragment>
      <h3 className={styles.header}>Search Cars</h3>
      <div>
        <form className={styles.form}>
          <label htmlFor="searchTerm">Search: </label>
          <input type="text" value={searchTerm} onChange={updateSearchTerm} />
        </form>
      </div>
      <Cars cars={cars} />
    </Fragment>
  );
}
