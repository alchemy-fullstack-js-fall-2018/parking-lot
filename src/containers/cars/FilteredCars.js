import FilteredCars from '../../components/FilteredCars/FilteredCars';
import { connect } from 'react-redux';
import { updateCarsSearchTerm } from '../../actions/cars';
import { getFilteredCars, getCarsSearchTerm } from '../../selectors/cars';

const mapStateToProps = state => ({
  cars: getFilteredCars(state),
  searchTerm: getCarsSearchTerm(state),
  summary: 'Registered Cars'
});

const mapDispatchToProps = dispatch => ({
  updateSearchTerm: ({ target }) => dispatch(updateCarsSearchTerm(target.value))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FilteredCars);
