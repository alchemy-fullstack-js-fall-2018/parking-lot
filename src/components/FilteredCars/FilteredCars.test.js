import React from 'react';
import { shallow } from 'enzyme';
import FilteredCars from './FilteredCars';

jest.mock('../../routes');

describe ('tests for FilteredCars component', () => {
  const props = {
    summary: 'Test FilteredCars',
    searchTerm: 'Acclaim',
    cars: [{
      plate: 'ZZZ000',
      make: 'Plymouth',
      model:  'Edsel',
      color: 'pink',
      lotUseCount: 2,
      isInLot: true
    },
    {
      plate: 'YYY111',
      make: 'Plymouth',
      model:  'Acclaim',
      color: 'red',
      lotUseCount: 3,
      isInLot: false
    }]
  };
  const updateSearchTerm = jest.fn();

  it('renders a FilteredCars component', () => {
    const wrapper = shallow(<FilteredCars
      cars={props.cars}
      searchTerm={props.searchTerm}
      updateSearchTerm={updateSearchTerm}
      summary={props.summary}
    />);
    expect(wrapper).toMatchSnapshot();
  });

});
