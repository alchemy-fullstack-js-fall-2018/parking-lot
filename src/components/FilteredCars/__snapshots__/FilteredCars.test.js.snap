// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for FilteredCars component renders a FilteredCars component 1`] = `
<Fragment>
  <form>
    <label
      htmlFor="searchTerm"
    >
      Search: 
    </label>
    <input
      onChange={[MockFunction]}
      type="text"
      value="Acclaim"
    />
  </form>
  <Cars
    cars={
      Array [
        Object {
          "color": "pink",
          "isInLot": true,
          "lotUseCount": 2,
          "make": "Plymouth",
          "model": "Edsel",
          "plate": "ZZZ000",
        },
        Object {
          "color": "red",
          "isInLot": false,
          "lotUseCount": 3,
          "make": "Plymouth",
          "model": "Acclaim",
          "plate": "YYY111",
        },
      ]
    }
    summary="Test FilteredCars"
  />
</Fragment>
`;
